src/axom/primal/geometry/Tetrahedron.hpp

```diff
@@ -220,6 +220,23 @@ class Tetrahedron
     return res;
   }
 
+  /*! 
+   * \brief Returns whether point \a p is contained within the tetrahedron (within tolerance \a eps)
+   */
+  bool contains(const PointType& p, double eps = 1e-8) const
+  {
+    if(axom::utilities::isNearlyEqual(ppedVolume(), 0., eps))
+    {
+      return false;
+    }
+
+    const auto bC = physToBarycentric(p);
+
+    return (bC[0] >= (0.0 - eps)) && (bC[1] >= (0.0 - eps)) && (bC[2] >= (0.0 - eps)) &&
+      (bC[3] >= (0.0 - eps)) && (bC[0] <= (1.0 + eps)) && (bC[1] <= (1.0 + eps)) &&
+      (bC[2] <= (1.0 + eps)) && (bC[3] <= (1.0 + eps));
+  }
+
```

Comment:
Definitely not worth doing in this PR, but I wonder if the contains query would be significantly faster if checks were done as each barycentric coordinate is computed (so we could shortcircuit and avoid computing all the barycentric coordinates).

Reply:
I was thinking something along these lines. I'm working on a follow-up (#1549) that will likely benefit from this sort of optimization, but wanted to get this in place first.